src/custom_nodes/common/utils.hpp

```diff
@@ -56,3 +63,419 @@ bool get_buffer(CustomNodeLibraryInternalManager* internalManager, T** buffer, c
+            } catch (std::out_of_range& e) {
+                return defaultValue;
+            }
+        }
+    }
+    return defaultValue;
+}
+
+float get_float_parameter(const std::string& name, const struct CustomNodeParam* params, int paramsCount, bool& isDefined, float defaultValue = 0.0f) {
+    isDefined = true;
+    for (int i = 0; i < paramsCount; i++) {
+        if (name == params[i].key) {
+            try {
+                return std::stof(params[i].value);
+            } catch (std::invalid_argument& e) {
+                isDefined = false;
+                return defaultValue;
+            } catch (std::out_of_range& e) {
+                isDefined = false;
+                return defaultValue;
+            }
+        }
+    }
+    isDefined = false;
+    return defaultValue;
+}
+
+std::string get_string_parameter(const std::string& name, const struct CustomNodeParam* params, int paramsCount, const std::string& defaultValue = "") {
+    for (int i = 0; i < paramsCount; i++) {
+        if (name == params[i].key) {
+            return params[i].value;
+        }
+    }
+    return defaultValue;
+}
+
+std::vector<float> get_float_list_parameter(const std::string& name, const struct CustomNodeParam* params, int paramsCount) {
+    std::string listStr;
+    for (int i = 0; i < paramsCount; i++) {
+        if (name == params[i].key) {
+            listStr = params[i].value;
+            break;
+        }
+    }
+
+    if (listStr.length() < 2 || listStr.front() != '[' || listStr.back() != ']') {
+        return {};
+    }
+
+    listStr = listStr.substr(1, listStr.size() - 2);
+
+    std::vector<float> result;
+
+    std::stringstream lineStream(listStr);
+    std::string element;
+    while (std::getline(lineStream, element, ',')) {
+        try {
+            float e = std::stof(element.c_str());
+            result.push_back(e);
+        } catch (std::invalid_argument& e) {
+            NODE_EXPECT(false, "error parsing list parameter");
+            return {};
+        } catch (std::out_of_range& e) {
+            NODE_EXPECT(false, "error parsing list parameter");
+            return {};
+        }
+    }
+
+    return result;
+}
+
+std::string floatListToString(const std::vector<float>& values) {
+    std::stringstream ss;
+    ss << "[";
+    for (size_t i = 0; i < values.size(); ++i) {
+        if (i != 0)
+            ss << ",";
+        ss << values[i];
+    }
+    ss << "]";
+    return ss.str();
+}
+
+void cleanup(CustomNodeTensor& tensor, CustomNodeLibraryInternalManager* internalManager) {
+    release(tensor.data, internalManager);
+    release(tensor.dims, internalManager);
+}
+
+void cleanup(CustomNodeTensor& tensor) {
+    free(tensor.data);
+    free(tensor.dims);
+}
+
+bool copy_confidences_into_output(struct CustomNodeTensor* output, const std::vector<float>& confidences, const std::string& tensor_name, const std::string& dims_name, CustomNodeLibraryInternalManager* internalManager) {
```

Comment:
I don't think moving everything as is to util.hpp file is good solution. This could be re-made to support generic vector write to `CustomNodeTensor` and take destination shape into consideration as well instead of hardcoding [N, 1, 1].

Reply:
Otherwise the util.hpp will end up being our file we put everything into just to clear main custom node source file.

Reply:
In the long run we might design custom node utility library or at least separate component with friendly API for tensor and image operations.